src/packagers/SourcemapPackager.js

```diff
@@ -0,0 +1,74 @@
+const fs = require('fs');
+const path = require('path');
+const Packager = require('./Packager');
+const promisify = require('../utils/promisify');
+const urlJoin = require('../utils/urlJoin');
+const sourceMap = require('source-map');
+
+class SourcemapPackager extends Packager {
+  setup() {
+    this.location = this.bundle.name + '.map';
+    this.dest = fs.createWriteStream(this.location);
+    this.dest.write = promisify(this.dest.write.bind(this.dest));
+    this.dest.end = promisify(this.dest.end.bind(this.dest));
+
+    this.generator = new sourceMap.SourceMapGenerator({
+      file: path.basename(this.bundle.name)
+    });
+    this.url = urlJoin(this.options.publicURL, path.basename(this.location));
+  }
+
+  async addAsset(asset) {
+    if (asset.generated.map) {
+      try {
+        await this.addMap(asset.generated.map);
+      } catch (e) {
+        // console.log(e);
+      }
+    }
+  }
+
+  async addMap(map) {
+    let inputMapConsumer = map.computeColumnSpans
+      ? map
+      : new sourceMap.SourceMapConsumer(map);
+    let addedSources = {};
+
+    // Add all mappings from asset to bundle
+    inputMapConsumer.eachMapping(mapping => {
+      if (!mapping.source) {
+        return false;
+      }
+      this.generator.addMapping({
+        source: mapping.source,
+        original: {
+          line: mapping.originalLine,
+          column: mapping.originalColumn
+        },
+        generated: {
+          line: mapping.generatedLine,
```

Comment:
don't we need to offset this line by the current line number in the file since we're concatenating a bunch of things together? and what about the extra lines added by the JSPackager... hmm

Reply:
We probably should, i'll dive deeper into this